next/unittests/databaseapi_test.py

```diff
@@ -0,0 +1,250 @@
+    db.client.close()
+
+# === basic tests ===
+def test_connect(db):
+    assert db.is_connected()
+
+def test_reconnection(db):
+    db.connect_mongo(MONGO_HOST, MONGO_PORT)
+    assert db.is_connected()
+
+# === test db functions ===
+def test__bucket(db):
+    assert db._bucket('foo') == db.client[MONGO_DB]['foo']
+
+def test_exists(db):
+    B = 'test_exists'
+
+    doc_uid = db._bucket(B).insert_one({'a_key': 2}).inserted_id
+
+    assert db.exists(B, doc_uid, 'a_key')
+    assert not db.exists(B, doc_uid, 'a_nonexistent_key')
+    assert not db.exists(B, 'ashkjfdbkjfns', 'a_key')
+
+def test_get(db):
+    B = 'test_get'
+
+    doc_uid = db._bucket(B).insert_one({'a_key': 2, 'another_key': [1.0, 'f']}).inserted_id
+
+    assert db.get(B, doc_uid, 'a_key') == 2
+    assert db.get(B, doc_uid, 'another_key') == [1.0, 'f']
+    assert not db.get(B, doc_uid, 'a_nonexistant_key')
+
+def test_get_many(db):
+    B = 'test_get_many'
+    doc = {'a_key': 2, 'another_key': [1.0, 'f'], 'third_key': 'baz'}
+
+    doc_uid = db._bucket(B).insert_one(doc).inserted_id
+
+    assert db.get_many(B, doc_uid, ['another_key']) \
+        == {'another_key': [1.0, 'f']}
+    assert db.get_many(B, doc_uid, ['another_key', 'third_key']) \
+        == {'another_key': [1.0, 'f'], 'third_key': 'baz'}
+
+    assert db.get_many(B, doc_uid, ['totally_nonexistent_key', 'another_key']) \
+        == {'totally_nonexistent_key': None, 'another_key': [1.0, 'f']}
+
+def test_get_and_delete(db):
+    B = 'test_get_and_delete'
+    doc = {'a': 2, 'b': [1.0, 'f'], 'c': 'baz'}
+
+    doc_uid = db._bucket(B).insert_one(doc).inserted_id
+
+    assert db.exists(B, doc_uid, 'b')
+    assert db.get_and_delete(B, doc_uid, 'b') == [1.0, 'f']
+    assert not db.exists(B, doc_uid, 'b')
+
+def test_increment(db):
+    B = 'test_increment'
+    
+    doc_uid = db._bucket(B).insert_one({'a': 2}).inserted_id
+
+    assert db.increment(B, doc_uid, 'a') == 3
+    assert db.get(B, doc_uid, 'a') == 3
+    assert db.increment(B, doc_uid, 'a', -2) == 1
+    assert db.get(B, doc_uid, 'a') == 1
+
+def test_increment_many(db):
+    B = 'test_increment_many'
+    doc = {'a': 0, 'b': 0, 'c': 0}
+    
+    doc_uid = db._bucket(B).insert_one(doc).inserted_id
+
+    assert db.increment_many(B, doc_uid, {'a': 1, 'b': 5, 'c': -7}) \
+                    == {'a': 1, 'b': 5, 'c': -7}
+    assert db.get_many(B, doc_uid, ['a', 'b', 'c']) \
+                    == {'a': 1, 'b': 5, 'c': -7}
+
+def test_pop_list(db):
+    B = 'test_pop_list'
+    doc = {'a': range(0, 10+1)}
+
+    doc_uid = db._bucket(B).insert_one(doc).inserted_id
+
+    assert db.pop_list(B, doc_uid, 'a', -1) == 10
+    assert db.get(B, doc_uid, 'a') == range(0, 9+1)
+
+    assert db.pop_list(B, doc_uid, 'a', 0) == 0
+    assert db.get(B, doc_uid, 'a') == range(1, 9+1)
+
+def test_append_list(db):
+    B = 'test_pop_list'
+
+    doc_uid = db._bucket(B).insert_one({'a': [1, 2, 3, 4]}).inserted_id
+
+    db.append_list(B, doc_uid, 'a', 10)
+
```

Comment:
I'm not opposed to returning the appended list.  I see no reason to strictly adhere to the python api exactly.  Seems kind of dumb/unintuitive to force people to `get` and do a separate `append`.